check.py

```diff
@@ -592,6 +593,16 @@ def run_gcc_tests():
       fail_if_not_identical_to_file(actual, expected)
 
 
+def run_unittest():
+  global num_failures
```

Comment:
Hmm.. this seems odd.  num_failures is a variable that is imported from  `scripts.test.shared` but is a global in this file.     Is this doing what you expect it to do?  If we want changes to be visible in shared we would probably not use `global` by `shared.num_failures`.

Reply:
Looking at the details of how `num_failures` is currently used, I think you probably want to use a new local here.. either that or somehow hook into the existing `shared.fail_with_error` system.

Reply:
Yes, this is definitely doing what I expect it to do, which is count unittest failures as equivalent to any other failures. `shared.num_failures` is imported into the global namespace of this module, but `shared` is not, so using `shared.num_failures` instead of `global num_failures` would require a redundant import. Looking at `shared.fail_with_error`, it basically does the same thing but respects the `abort_on_first_failure` option. I will update this code to respect that option as well.

Reply:
But the num_failures that is global the this module is different to one in shared isn't it? 

Updating num_failures in shared won't effect this value and vice versa.  I guess that bug already existed but this make it worse.

Reply:
In python when you re-assign a global, it doesn't effect modules that have already imported that global.

Reply:
Wow you're totally right. That's nuts. TIL. Fixing now.

Reply:
... I take back some of the nice things I've said about Python